piccolo/columns/m2m.py

```diff
@@ -0,0 +1,336 @@
+
+    def __await__(self):
+        return self.run().__await__()
+
+
+@dataclass
+class M2MRemoveRelated:
+
+    target_row: Table
+    m2m: M2M
+    rows: t.Sequence[Table]
+
+    async def run(self):
+        fk = self.m2m._meta.secondary_foreign_key
+        related_table = fk._foreign_key_meta.resolved_references
+
+        row_ids = []
+
+        for row in self.rows:
+            if row.__class__ != related_table:
+                raise ValueError("The row belongs to the wrong table!")
+
+            row_id = getattr(row, row._meta.primary_key._meta.name)
+            if row_id:
+                row_ids.append(row_id)
+
+        if row_ids:
+            return (
+                await self.m2m._meta.resolved_joining_table.delete()
+                .where(
+                    self.m2m._meta.primary_foreign_key == self.target_row,
+                    self.m2m._meta.secondary_foreign_key.is_in(row_ids),
+                )
+                .run()
+            )
+
+        return None
+
+    def run_sync(self):
+        return run_sync(self.run())
+
+    def __await__(self):
+        return self.run().__await__()
+
+
+@dataclass
+class M2MGetRelated:
+
+    row: Table
+    m2m: M2M
+
+    async def run(self):
+        joining_table = self.m2m._meta.resolved_joining_table
+
+        async with self.row._meta.db.transaction():
+            secondary_table = (
+                self.m2m._meta.secondary_foreign_key._foreign_key_meta.resolved_references  # noqa: E501
+            )
+
+            ids = (
+                await joining_table.select(
+                    getattr(
+                        self.m2m._meta.secondary_foreign_key,
+                        secondary_table._meta.primary_key._meta.name,
+                    )
+                )
+                .where(self.m2m._meta.primary_foreign_key == self.row)
+                .output(as_list=True)
+            )
+
+            results = await secondary_table.objects().where(
+                secondary_table._meta.primary_key.is_in(ids)
+            )
+
+            return results
+
+    def run_sync(self):
+        return run_sync(self.run())
+
+    def __await__(self):
+        return self.run().__await__()
+
+
+class M2M:
+    def __init__(
+        self,
+        joining_table: t.Union[t.Type[Table], LazyTableReference],
+        foreign_key_columns: t.Optional[t.List[ForeignKey]] = None,
+    ):
+        """
+        :param joining_table:
+            A ``Table`` containing two ``ForeignKey`` columns.
+        :param foreign_key_columns:
+            If for some reason your joining table has more than two foreign key
+            columns, you can explicitly specify which two are relevant.
+
+        """
+        if foreign_key_columns:
+            if len(foreign_key_columns) != 2 or not all(
+                isinstance(i, ForeignKey) for i in foreign_key_columns
+            ):
+                raise ValueError("You must specify two ForeignKey columns.")
```

Comment:
```suggestion
        if foreign_key_columns and (
            len(foreign_key_columns) != 2
            or not all(isinstance(i, ForeignKey) for i in foreign_key_columns)
        ):
            raise ValueError("You must specify two ForeignKey columns.")
```

Too much nesting can make code difficult to understand, and this is especially true in Python, where there are no brackets to help out with the delineation of different nesting levels.

Reading deeply nested code is confusing since you have to keep track of which conditions relate to which levels.